pkg/credentialprovider/gcp/metadata.go

```diff
@@ -148,18 +166,73 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 	return credentialprovider.DockerConfig{}
 }
 
+func runWithBackoff(url string, f func() ([]byte, error)) ([]byte, error) {
```

Comment:
We can add a large timeout if infinite loop appears to intimidating. The question is that if kubelet comes up with metadata server not reachable at all, would it encounter other problems (e.g., dns). I'd prefer kubelet just *fail* and exit in this case, but it's strange to exit the process in this function.